core/domain/exp_jobs_one_off.py

```diff
@@ -216,69 +260,148 @@ def map(item):
+    along with the validity and type (expression/equation) of the inputs present
+    in the exploration.
+
+    This validation is done by the 'is_valid_math_expression' or the
+    'is_valid_math_equation' function present in schema_utils.py.
+    """
+
+    @classmethod
+    def entity_classes_to_map_over(cls):
+        return [exp_models.ExplorationModel]
+
+    @staticmethod
+    def map(item):
+        is_valid_math_expression = schema_utils.get_validator(
+            'is_valid_math_expression')
+        is_valid_math_equation = schema_utils.get_validator(
+            'is_valid_math_equation')
+        ltt = latex2text.LatexNodes2Text()
+        committer_id = 'OppiaMigrationBot'
+
+        if not item.deleted:
+            exploration = exp_fetchers.get_exploration_from_model(item)
+            for state_name, state in exploration.states.items():
+                if state.interaction.id == 'MathExpressionInput':
+                    new_answer_groups = []
+                    new_interaction_id = ''
+                    for group in state.interaction.answer_groups:
+                        new_answer_group = copy.deepcopy(group)
+                        for rule_spec in new_answer_group.rule_specs:
+                            rule_input = ltt.latex_to_text(
+                                rule_spec.inputs['x'])
+
+                            rule_input = _clean_math_expression(rule_input)
+
+                            validity = _INVALID_EXPR
+                            if is_valid_math_expression(
+                                    rule_input, algebraic=True):
+                                validity = _VALID_ALG_EXPR
+                                new_interaction_id = 'AlgebraicExpressionInput'
+                            elif is_valid_math_expression(
+                                    rule_input, algebraic=False):
+                                validity = _VALID_NUM_EXPR
+                                new_interaction_id = 'NumericExpressionInput'
                             elif is_valid_math_equation(rule_input):
-                                validity = 'Valid Equation'
+                                validity = _VALID_MATH_EQN
+                                new_interaction_id = 'MathEquationInput'
+
+                            if validity != _INVALID_EXPR:
+                                rule_spec.inputs['x'] = rule_input
+                                if validity == _VALID_MATH_EQN:
+                                    rule_spec.inputs['y'] = 'both'
+                                rule_spec.rule_type = 'MatchesExactlyWith'
 
                             output_values = '%s %s: %s' % (
                                 item.id, state_name, rule_input)
 
                             yield (validity, output_values.encode('utf-8'))
 
+                        new_answer_groups.append(new_answer_group.to_dict())
+
+                    if new_interaction_id != '':
```

Comment:
if new_interaction_id does become '', then that means, it didn't fall into any one case above, right? Handle that as well, by raising  an error?

Reply:
I've changed the code so that it is consistent in both the jobs and easier to follow, which is why this is not required.